### ISSUE TITLE
Error on the external secret during ci-knative-eventing-s390x-e2e-tests

### ISSUE THREAD

**Issue by BbolroC** [GoogleCloudPlatform/oss-test-infra#866]
Got the following error after merging #843:

https://storage.googleapis.com/knative-prow/logs/ci-knative-eventing-s390x-e2e-tests/1395650582487764992/build-log.txt

It seems like there is a problem of loading the secret `s390x-cluster1`.

@chizhg Could you please go over this and suggest how to fix the problem?

Thank you.

**Comment by BbolroC** [comment]
To ease the investigation, the secrets in the secret manager follow the IAM policies below:

```
~ » gcloud secrets get-iam-policy kubeconfig
bindings:
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.secretAccessor
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.viewer
etag: BwXCDDOsGsc=
version: 1
~ » gcloud secrets get-iam-policy ko-docker-repository
bindings:
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.secretAccessor
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.viewer
etag: BwXCDDqRIDA=
version: 1
~ » gcloud secrets get-iam-policy registry-certificate
bindings:
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.secretAccessor
- members:
  - serviceAccount:kubernetes-external-secrets-sa@knative-tests.iam.gserviceaccount.com
  role: roles/secretmanager.viewer
etag: BwXCDDza9_4=
version: 1
```

**Comment by chaodaiG** [comment]
I just inspected the build cluster with `k -ntest-pods get secrets s390x-cluster1 -oyaml`, and the secret was reconciled by external secret:
```
metadata:
  creationTimestamp: "2021-05-20T23:59:52Z"
  name: s390x-cluster1
  namespace: test-pods
  ownerReferences:
  - apiVersion: kubernetes-client.io/v1
    controller: true
    kind: ExternalSecret
    name: s390x-cluster1
```

**Comment by BbolroC** [comment]
Thank you very much for the check.

Could you also confirm the following:

1. if the secret (kind: `Secret`) is accordingly created by the `ExternalSecret`?

2. if the namespace where the deleted secret resided is `test-pods`?

**Comment by chaodaiG** [comment]
> if the namespace where the deleted secret resided is test-pods?

Actually I'm not sure about this, @chizhg would know about the deleted secret

**Comment by chizhg** [comment]
Yes, it was. The command I ran was `k delete secret s390x-cluster1 -n test-pods`.

**Comment by BbolroC** [comment]
Okay, then I will figure out what I can do to solve the problem from my side. I would like to leave this issue open until the cause is identified. Thanks for the confirmation. :wink:

**Comment by BbolroC** [comment]
> Yeah it was, the full yaml is:
> 
> ```
> apiVersion: v1
> data:
>   <hidden>
> kind: Secret
> metadata:
>   creationTimestamp: "2021-05-20T23:59:52Z"
>   name: s390x-cluster1
>   namespace: test-pods
>   ownerReferences:
>   - apiVersion: kubernetes-client.io/v1
>     controller: true
>     kind: ExternalSecret
>     name: s390x-cluster1
> type: Opaque
> ```
> 
> And yes it's in `test-pods` ns as the command includes `-ntest-pods`

Sorry to bother you again. Would you mind sending me the hidden data part via email? I would like to compare it to the data working before. My email address is <bbolroc@gmail.com>. Thank you. ^_^

**Comment by chaodaiG** [comment]
That doesn't feel very secure. I think I have figured out why. I have manually inspected one of the value of the secret, and when I do ` k --context=knative-prow-builds -ntest-pods get secrets s390x-cluster1 -ojsonpath="{.data.ko-docker-repo}" | base64 -d` the result was still base64 encoded, and when I run `base64 -d` again on the string the output was a reasonable ip address. 

The solution is: store base64 decoded value in secret manager instead, and the cluster will encode it for you
